### PR TITLE
Support Mac arm64

### PR DESCRIPTION
**Pull request by CzBiX**
#### Summary
Build "Support Mac arm64 build"

#### Purpose of change

Get native arm64 build for mac.

#### Describe the solution

Use framework for mac to avoid cross compile.

### CODE REVIEW

**Comment by BrettDong** [comment]
As far as I know, an ARM binary compiled on a different machine won't run unless code signed with a valid Apple developer signature.

**Comment by CzBiX** [comment]
yes, but there are two things.
The linker will automatically ad-hoc signing output binary, and which then can be run by "Open" in the context menu.
While I believe this is basic knowledge of using open source software on a mac, we are supposed to add it to the FAQ.
see: https://support.apple.com/guide/mac-help/open-a-mac-app-from-an-unidentified-developer-mh40616/mac

**Comment by BrettDong** [comment]
What I was saying is that, the ad-hoc signature generated on machine A (in this case, GitHub Actions servers) is not accepted on another machine B (user's own Mac). When the user tries to run the binary on machine B, it will immediately be killed by the kernel (zsh: Killed). I'm not talking about the quarantine flag which can be workaround by right-click and select "open".

**Comment by CzBiX** [comment]
I get it, I've confirmed it works.
if you have the ARM mac, I can upload what I built to you, so you can also verify it.

**Comment by BrettDong** [comment]
If you confirm the binary compiled on GitHub Actions can run on a local ARM Mac then never mind about what I said. I don't have ARM Mac at hand now. When M1 just came out I got some and I could remember an ad-hoc signature is only valid on the same machine. Apple might have updated the security policy at some point later.

**Comment by kevingranade** [comment]
See https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/4271800540 for tge build errors.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -262,8 +271,12 @@ jobs:
       - name: Install dependencies (mac)
         if: runner.os == 'macOS'
         run: |
-          HOMEBREW_NO_AUTO_UPDATE=yes HOMEBREW_NO_INSTALL_CLEANUP=yes brew install sdl2 sdl2_image sdl2_ttf sdl2_mixer gettext ccache parallel
+          brew install gettext ccache parallel
+          ~/build-script/get-mac-sdl2-framework.sh
```
Comment (kebkaldanil) [inline .github/workflows/release.yml:275]: typo: ~~build-script~~ **build-scripts**

Comment (LyleSY) [reply .github/workflows/release.yml:275]: Yep, no such file or directory, this is it

**Pull request merged**